### ISSUE TITLE
ddterm is killed after locking screen

### ISSUE THREAD

**Issue by magnetik** [ddterm/gnome-shell-extension-ddterm#1087]
### GNOME Shell version

GNOME Shell 46.0

### Linux distro

"Ubuntu 24.04.1 LTS"

### ddterm version

ddterm 55 revision 28b17841346f9cf1519138c88586168183bea29c
Extension 55 revision 28b17841346f9cf1519138c88586168183bea29

### Session type

Wayland (`echo $WAYLAND_DISPLAY` output is non-empty)

### Special configuration

- [ ] High DPI
- [X] More than one monitor

### All installed extensions

```Text
gnome-wireguard-extension@SJBERTRAND.github.com
  Nom: WireGuard-VPN-extension
  Description: Enable/Disable Wireguard tunnels
  Chemin: /home/baptiste/.local/share/gnome-shell/extensions/gnome-wireguard-extension@SJBERTRAND.github.com
  URL: https://github.com/SJBERTRAND/gnome-wireguard-extension
  Version: 14
  Activé: Non
  État: INITIALIZED

dash-to-panel@jderose9.github.com
  Nom: Dash to Panel
  Description: An icon taskbar for the Gnome Shell. This extension moves the dash into the gnome main panel so that the application launchers and system tray are combined into a single panel, similar to that found in KDE Plasma and Windows 7+. A separate dock is no longer needed for easy access to running and favorited applications.

For a more traditional experience, you may also want to use Tweak Tool to enable Windows > Titlebar Buttons > Minimize & Maximize.

For the best support, please report any issues on Github. Dash-to-panel is developed and maintained by @jderose9 and @charlesg99.
  Chemin: /home/baptiste/.local/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com
  URL: https://github.com/home-sweet-gnome/dash-to-panel
  Version: 64
  Activé: Oui
  État: ACTIVE

clipboard-indicator@tudmotu.com
  Nom: Clipboard Indicator
  Description: The most popular clipboard manager for GNOME, with over 1M downloads
  Chemin: /home/baptiste/.local/share/gnome-shell/extensions/clipboard-indicator@tudmotu.com
  URL: https://github.com/Tudmotu/gnome-shell-extension-clipboard-indicator
  Version: 65
  Activé: Oui
  État: ACTIVE

ddterm@amezin.github.com
  Nom: ddterm
  Description: Another drop down terminal extension for GNOME Shell. With tabs. Works on Wayland natively
  Chemin: /home/baptiste/.local/share/gnome-shell/extensions/ddterm@amezin.github.com
  URL: https://github.com/ddterm/gnome-shell-extension-ddterm
  Version: 55
  Activé: Oui
  État: ACTIVE

BingWallpaper@ineffable-gmail.com
  Nom: Bing Wallpaper
  Description: Sync your wallpaper to today's Microsoft Bing image of the day (the image you see when you visit Bing.com).

 *Disclaimer*: this extension is unofficial and not affiliated with Bing or Microsoft in any way. Images are protected by copyright and are licensed only for use as wallpapers.

Features:
* UHD resolution wallpapers
* Automatically fetches current Bing wallpaper of the day and sets as both lock screen and desktop wallpaper (user selectable on GNOME versions that support it)
* Doesn't poll continuously - only once per day and on startup (schedules a refresh when Bing is due to update)
 * random mode (from previously downloaded wallpapers)
 *NEW: select/cycle wallpaper through previously downloaded images
* Language support: English (en), German (de), Dutch (nl), Italian (it), Polish (pl), Chinese (zh_CN, zh_TW), French (fr), Portuguese (pt), Ukrainian (uk), Russian (ru_RU), Spanish (es), Korean (ko), Indonesian (id), Catalan (ca), Norwegian Bokmål (nb) & Nynorsk (ni), Swedish (sv), Arabic (ar), Hungarian (hu) and Japanese (ja) - a HUGE thanks to the translators

This extension was forked from the NASA APOD extension by Elinvention (https://github.com/Elinvention) and inspired by Bing Desktop Wallpaper Changer by Utkarsh Gupta (https://github.com/UtkarshGpta).

Always restart GNOME after manually updating extensions. Please report bugs to the GitHub page below:
  Chemin: /home/baptiste/.local/share/gnome-shell/extensions/BingWallpaper@ineffable-gmail.com
  URL: https://github.com/neffo/bing-wallpaper-gnome-extension
  Version: 50
  Activé: Oui
  État: ACTIVE

nothing-to-say@extensions.gnome.wouter.bolsterl.ee
  Nom: Nothing to say
  Description: Unmute the microphone only when you have something to say.
  Chemin: /home/baptiste/.local/share/gnome-shell/extensions/nothing-to-say@extensions.gnome.wouter.bolsterl.ee
  URL: https://github.com/wbolster/nothing-to-say
  Version: 22
  Activé: Oui
  État: ACTIVE

Vitals@CoreCoding.com
  Nom: Vitals
  Description: A glimpse into your computer's temperature, voltage, fan speed, memory usage, processor load, system resources, network speed and storage stats. This is a one stop shop to monitor all of your vital sensors. Uses asynchronous polling to provide a smooth user experience. Feature requests or bugs? Please use GitHub.
  Chemin: /home/baptiste/.local/share/gnome-shell/extensions/Vitals@CoreCoding.com
  URL: https://github.com/corecoding/Vitals
  Version: 69
  Activé: Non
  État: INITIALIZED

emoji-copy@felipeftn
  Nom: Emoji Copy
  Description: Emoji copy is a versatile extension designed to simplify emoji selection and clipboard management.

It is a fork of Emoji Selector.
  Chemin: /home/baptiste/.local/share/gnome-shell/extensions/emoji-copy@felipeftn
  URL: https://github.com/felipeftn/emoji-copy
  Version: 25
  Activé: Oui
  État: ACTIVE

ding@rastersoft.com
  Nom: Desktop Icons NG (DING)
  Description: Adds icons to the desktop. Fork of the original Desktop Icons extension, with several enhancements .
  Chemin: /usr/share/gnome-shell/extensions/ding@rastersoft.com
  URL: https://gitlab.com/rastersoft/desktop-icons-ng
  Activé: Non
  État: INITIALIZED

tiling-assistant@ubuntu.com
  Nom: Ubuntu Tiling Assistant
  Description: Expand GNOME's 2 column tiling and add a Windows-snap-assist-inspired popup...
  Chemin: /usr/share/gnome-shell/extensions/tiling-assistant@ubuntu.com
  URL: https://launchpad.net/ubuntu/+source/gnome-shell-extension-tiling-assistant
  Version: 46
  Activé: Non
  État: INITIALIZED

ubuntu-appindicators@ubuntu.com
  Nom: Ubuntu AppIndicators
  Description: Support app indicators and legacy tray icons in top panel, as the default Ubuntu experience. System update of AppIndicator and KStatusNotifierItem Support extension.
  Chemin: /usr/share/gnome-shell/extensions/ubuntu-appindicators@ubuntu.com
  URL: https://github.com/ubuntu/gnome-shell-extension-appindicator
  Activé: Non
  État: INITIALIZED

ubuntu-dock@ubuntu.com
  Nom: Ubuntu Dock
  Description: A dock for the Gnome Shell, default Ubuntu experience. This extension is a modified version of Dash To Dock with different defaults. Dash To Dock can be installed to replace it and give more (but unsupported) configuration options.
  Chemin: /usr/share/gnome-shell/extensions/ubuntu-dock@ubuntu.com
  URL: https://micheleg.github.io/dash-to-dock/
  Auteur d’origine: micxgx@gmail.com
  Version: 90
  Activé: Non
  État: INACTIVE
```

### ddterm settings dump

```ini
[/]
background-color='rgb(253,246,227)'
cursor-blink-mode='off'
delete-binding='tty'
foreground-color='rgb(101,123,131)'
panel-icon-type='none'
show-animation='ease-in-quad'
tab-expand=false
tab-label-width=0.10000000000000001
use-theme-colors=false
window-maximize=false
window-size=0.49856321839080459
```

### Description

After locking screen (either after inactivity or with SUPER + L) and relog-in, ddterm is always killed. I was not hapening before. 

logs : 
```
oct. 23 18:47:47 baptiste-XPS-13-9310 gnome-shell[4002]: no authentication service is enabled, using password authentication
oct. 23 18:47:47 baptiste-XPS-13-9310 gnome-shell[4002]: no authentication service is enabled, using password authentication
oct. 23 18:47:50 baptiste-XPS-13-9310 gdm-password][244962]: gkr-pam: unlocked login keyring
oct. 23 18:47:50 baptiste-XPS-13-9310 gnome-shell[4002]: no authentication service is enabled, using password authentication
oct. 23 18:47:50 baptiste-XPS-13-9310 gnome-shell[4002]: no authentication service is enabled, using password authentication
oct. 23 18:47:50 baptiste-XPS-13-9310 dbus-daemon[3762]: [session uid=1000 pid=3762] Activating service name='org.freedesktop.FileManager1' requested by ':1.30' (uid=1000 pid=4002 comm="/usr/bin/gnome-shell" label="unconfined")
oct. 23 18:47:51 baptiste-XPS-13-9310 nautilus[244991]: Connecting to org.freedesktop.Tracker3.Miner.Files
oct. 23 18:47:51 baptiste-XPS-13-9310 dbus-daemon[3762]: [session uid=1000 pid=3762] Successfully activated service 'org.freedesktop.FileManager1'
oct. 23 18:47:51 baptiste-XPS-13-9310 gnome-shell[4002]: clutter_actor_remove_child: assertion 'child->priv->parent == self' failed
oct. 23 18:47:51 baptiste-XPS-13-9310 systemd[3726]: app-gnome-com.github.amezin.ddterm-102311.scope: Consumed 22.834s CPU time.
oct. 23 18:47:51 baptiste-XPS-13-9310 systemd[3726]: vte-spawn-f617db5e-cd54-4b3a-b287-fcaeead4418d.scope: Consumed 9.126s CPU time.
oct. 23 18:47:51 baptiste-XPS-13-9310 tracker-miner-fs-3[245012]: (tracker-extract-3:245012): GLib-GIO-WARNING **: 18:47:51.315: Error creating IO channel for /proc/self/mountinfo: Argument invalide (g-io-error-quark, 13)
oct. 23 18:47:51 baptiste-XPS-13-9310 dbus-daemon[3762]: [session uid=1000 pid=3762] Reloaded configuration
oct. 23 18:47:51 baptiste-XPS-13-9310 gnome-shell[4002]: /home/baptiste/.local/share/gnome-shell/extensions/ddterm@amezin.github.com/bin/com.github.amezin.ddterm: GLib.SpawnError: Le processus fils a été tué par le signal 15
                                                         
                                                         Stack trace:
                                                           wait_check/</<@file:///home/baptiste/.local/share/gnome-shell/extensions/ddterm@amezin.github.com/ddterm/shell/subprocess.js:231:36
                                                           @resource:///org/gnome/shell/ui/init.js:21:20
```

**Comment by amezin** [comment]
Ok this makes more sense now.

Extensions are normally disabled during screen lock (unless they have some specific metadata - ddterm doesn't). And ddterm extension shuts down the app when the extension is disabled. However, it should also detect that the extension is being disabled because of screen lock, and not terminate the app in this case. Guess it's not working as expected on your machine for some reason.

The bad news, however, is that I haven't been able to reproduce the bug so far.

**Comment by amezin** [comment]
Seems to be a compatibility issue with Dash to Panel. Apparently because Dash to Panel tries to enable/disable Ubuntu Dock from its own enable()/disable()

**Comment by amezin** [comment]
https://github.com/GNOME/gnome-shell/blob/bab37d3286710a0eab02866cf70aa5da69254b7b/js/ui/extensionSystem.js#L194-L200

Disabling one extension can trigger disabling of other extensions. So when Dash to Panel disables Ubuntu Dock, it also disables and reenables ddterm - which kills the app, because disable() isn't triggered by screen lock.

It's not obvious how can I determine whether ddterm is being disabled only temporarily - because some other extension is disabled, or if the user disables ddterm intentionally.

**Comment by magnetik** [comment]
Hey. Sorry for the ping but do you have an idea on when a release will include this fix? 
Thanks again for the work on this extension :+1:

**Comment by amezin** [comment]
Waiting for https://github.com/ddterm/gnome-shell-extension-ddterm/pull/1113 one more week

In the meantime you can [install](https://github.com/ddterm/gnome-shell-extension-ddterm/blob/master/docs/INSTALL.md#1a-prebuilt-package) the [nightly build](https://nightly.link/ddterm/gnome-shell-extension-ddterm/workflows/push/master/pack.zip)

**Comment by magnetik** [comment]
The nightly build is working perfectly, thanks !
